### PR TITLE
WIP: Add seaography documentation

### PR DESCRIPTION
**Pull request by karatakis**
## PR Info

Adds documentation for the new project of the SeaQL family Seaography.
Seaography is a CLI tool that generates a ready to compile Rust GraphQL API for a relational database.

The initial draft can be found here: https://github.com/SeaQL/summer-of-code/discussions/12

- Closes https://github.com/SeaQL/seaography/issues/20

### CODE REVIEW

**Comment by tyt2y3** [comment]
How can I access the preview?

**Comment by tyt2y3** [comment]
Seems like because there is a conflict, I just can't see the preview?

**Comment by tyt2y3** [comment]
Also, why doesn't this link work? https://deploy-preview-41--seaql.netlify.app/Seaography

**Comment by tyt2y3** [comment]
It seems like docusaurus is not correctly setup?

File: Seaography/docs/01-index.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  A runtime for writing reliable, asynchronous, and slim applications with the Rust programming language.
+
+5. [itertools*](https://docs.rs/itertools/latest/itertools/)
+
+  Extra iterator adaptors, functions and macros.
+
+* Recommended to study, but not required
+
+## How to use
```
Comment (tyt2y3) [inline Seaography/docs/01-index.md:54]: I am hoping to see a more comprehensive step by step guide, from the sample database to running GraphQL queries, perhaps with a few screenshots along the way if that help illustrate. Actually, if you have the energy, a short video screencast would be perfect! (It's not necessary though)

But yeah at least you get the idea

File: Seaography/docs/04-enchantments-cookbook.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+to
+
+```rust
+
+let schema = Schema::build(QueryRoot, EmptyMutation, EmptySubscription)
+    .data(database)
+    .data(orm_dataloader)
+    .limit_complexity(5)
```
Comment (tyt2y3) [inline Seaography/docs/04-enchantments-cookbook.md:103]: I think we *always* want to limit the complexity in practice. So I think the codegen should always generate this clause. And actually we can allow the user to config this number via the cli parameter.

**Comment by tyt2y3** [comment]
Also, I think we need one chapter discussing scaler filter and pagination, with a GraphQL input and a SQL output (sounds a lot like the test suite?)

And another one for relational query. We should first select 2 or 3 tables with relation from the sample database and make several GraphQL queries against them.

**Comment by tyt2y3** [comment]
Finally, I think we need a chapter for some hints to how to customize the query mechanism and/or extend the GraphQL schema.

I think we can also give a gentle introduction of GraphQL, how GraphQL resolvers work and perhaps comparison with other related GraphQL frameworks / engine.

**Comment by karatakis** [comment]
> Also, why doesn't this link work? https://deploy-preview-41--seaql.netlify.app/Seaography
> 
> It seems like docusaurus is not correctly set up?

Because the netlify build command is not configured correct

The current build command:
https://app.netlify.com/sites/seaql/deploys/62dd5353b844490009d00f68#L64

How to fix it:
* Extend build command to build Seaography too
* More info https://docs.netlify.com/configure-builds/overview/

File: Seaography/docs/01-index.md (hunk 1 above)
Comment (karatakis) [reply Seaography/docs/01-index.md:54]: Great point.

I will create a page "How to run the example database" and document the step by step process using pictures. 

Regarding the video I will take a note of it and work on it on the future.

File: Seaography/docs/04-enchantments-cookbook.md (hunk 2 above)
Comment (karatakis) [reply Seaography/docs/04-enchantments-cookbook.md:103]: I can add it and be CLI driven

**Comment by billy1624** [comment]
Hey @karatakis, I just make Netlify and GitHub Action to build docs using `build.sh` script.
- https://github.com/SeaQL/seaql.github.io/pull/46

You can add seaography build script into `build.sh` once and make seaography built in development and production environment. Please merge master into current PR.
Thanks!! :D

**Comment by karatakis** [comment]
> Also, I think we need one chapter discussing scaler filter and pagination, with a GraphQL input and a SQL output (sounds a lot like the test suite?)
> 
> And another chapter for relational query. We should first select 2 or 3 tables with relation from the sample database and make several GraphQL queries against them.

I will try to give some details in the example page

**Comment by karatakis** [comment]
> Finally, I think we need a chapter for some hints to how to customize the query mechanism and/or extend the GraphQL schema.
> 
> I think we can also give a gentle introduction of GraphQL, how GraphQL resolvers work and perhaps comparison with other related GraphQL frameworks / engine.

The "encancements-cookbook" is meant to be a chapter to explain how to extend the generated code. I will look to rename it and add more documentation on how to extend

File: Seaography/docs/05-running-example.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  cd generated
+  cargo run
+  ```
+
+5. Browse local server
+  http://127.0.0.1:8000/
+
+## Query examples
+
+### Fetch albums where artist ID = 1 || ID = 3
```
Comment (tyt2y3) [inline Seaography/docs/05-running-example.md:62]: Nice. I really liked these. This is where everything comes to live!

File: Seaography/docs/01-getting-started.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+4. [tokio*](https://docs.rs/tokio/latest/tokio/)
+
+  A runtime for writing reliable, asynchronous, and slim applications with the Rust programming language.
+
+5. [itertools*](https://docs.rs/itertools/latest/itertools/)
+
+  Extra iterator adaptors, functions and macros.
+
+* Recommended to study, but not required
```
Comment (billy1624) [inline Seaography/docs/01-getting-started.md:44]: The `*` makes this line being interpreted as a list. Consider escaping it :)

```suggestion
\* Recommended to study, but not required
```

**Comment by tyt2y3** [comment]
@karatakis Let me know when you want to publish an initial version of this

**Comment by tyt2y3** [comment]
@billy1624 how do we syntax highlight Rust?

image

**Pull request merged**